### PR TITLE
global: marshmallow 2 and 3 compatibility

### PR DESCRIPTION
**Pull request by kprzerwa**

### CODE REVIEW

File: invenio_files_rest/serializer.py [hunk 1]
```diff
@@ -18,6 +19,44 @@
...
+
+        def data_errors(self):
+            warnings.warn(
+                "Schema().dump().errors attribute is deprecated "
+                "in marshmallow v3.x",
+                category=DeprecationWarning, stacklevel=2)
+            return result
+
+        setattr(cls, 'data', property(data))
+        setattr(cls, 'errors', property(data_errors))
```
Comment (kprzerwa) [inline invenio_files_rest/serializer.py]: after a thought, I would remove this and let it fail on errors, or I would raise an exception, but I will wait for a feedback

File: setup.py [hunk 2]
```diff
@@ -63,11 +63,14 @@
     'Flask>=0.11.1',
     'fs>=0.5.4,<2.0',
     'invenio-rest[cors]>=1.0.0',
-    'marshmallow>=2.15.1,<3.0.0',
+    'marshmallow>=2.15.1,<3.0.0;python_version<"3.0.0"',
```
Comment (lnielsen) [inline setup.py]: Please see https://github.com/inveniosoftware/troubleshooting/issues/1

File: invenio_files_rest/serializer.py [hunk 3]
```diff
@@ -18,6 +19,44 @@
...
 
 
+class MarshalResultBase(object):
+    """Wrapper class to provide backward compatibility with marshmallow 2."""
+
+    def __new__(cls, result, errors=None, *args, **kwargs):
+        """Instanciate Marshal Result class."""
+        def data(self):
+            warnings.warn(
+                "Schema().dump().data attribute is deprecated "
```
Comment (lnielsen) [inline invenio_files_rest/serializer.py]: Will this issue the warning on Marshmallow 2? 

I think the warning should only be issued when v3 is installed to warn users that their schema will break at some point.

I guess this class is also useful for other modules. Perhaps we need to find another home for it?

**Comment by lnielsen** [comment]
Will you update once invenosoftware/invenio-rest#94 is done? I think it still makes sense to include marshmallow as a devel test.

**Pull request merged**